### PR TITLE
[3006.x] Draft: 65264 retain existing perms (RPM)

### PR DESCRIPTION
**Pull request by barneysowood**
### What issues does this PR fix or reference?
Fixes: #65264 

### New Behavior
Remove this section if not relevant

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [ ] Docs
- [ ] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [ ] Tests written/updated

### Commits signed with GPG?
Yes/No

Please review [Salt's Contributing Guide](https://docs.saltproject.io/en/master/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

File: pkg/rpm/salt.spec [hunk 1]
```diff
@@ -424,11 +424,22 @@ usermod -c "%{_SALT_NAME}" \
...
+
+  # if upgrading from 3006.0/3006.1/3006.2 reset permissions
+  # for salt user - issue 64193 and 65264
+
+  case "${INSTALLED_VERSION}" in
+    3006.0|3006.1|3006.2)
+      PY_VER=$(/opt/saltstack/salt/bin/python3 -c "import sys; sys.stdout.write('{}.{}'.format(*sys.version_info)); sys.stdout.flush();")
+      find /etc/salt /opt/saltstack/salt /var/log/salt /var/cache/salt /var/run/salt \
+        \! \( -path /etc/salt/cloud.deploy.d\* -o -path /var/log/salt/cloud -o -path /opt/saltstack/salt/lib/python${PY_VER}/site-packages/salt/cloud/deploy\* \) -a \
+        \( -user salt -o -group salt \) -exec chown root:root \{\} \;
```
Comment (dmurphy18) [inline pkg/rpm/salt.spec:438]: The machine might not be root ownership, what if they had installed as non-root. Can think of one customer which names it something completely different

**Review by dmurphy18 — CHANGES_REQUESTED** [review]
the previous owner may not be root, and more than 3006.0-2 at this point

**Review by dmurphy18 — COMMENTED** [review]
wondering about changes I made in https://github.com/saltstack/salt/pull/66218 and if applicable ?, That and not version specific, esp. since 3005.x is EOL

**Comment by dmurphy18** [comment]
@barneysowood Given this PR is not getting a lot of attention, wondering if you would be fine with my taking it over

**Comment by barneysowood** [comment]
> @barneysowood Given this PR is not getting a lot of attention, wondering if you would be fine with my taking it over

@dmurphy18 - sorry, I just haven't been able to spend any time on this in the last few weeks. If you're able to take over that would be great, thanks.

**Comment by dmurphy18** [comment]
Closing this in favor of PR https://github.com/saltstack/salt/pull/66218

**Pull request closed without merging**